### DISCUSSION TITLE
Inconsistent RTSP session IDs

### DISCUSSION THREAD

**Discussion by jesuscampos** [bluenviron/mediamtx#3357] [Questions] [answered]
### Question

Hi,

I am experiencing an issue with inconsistent RTSP session IDs when using MediaMTX as an RTSP server. Here is a brief overview of my setup and the problem:

### Setup:
MediaMTX Version: v1.3.1
Protocol: RTSP (TCP only)

### Problem:
An intermediate proxy captures the negotiated RTSP session and sends it to telemetry for issue tracking purposes. The hyphens have been removed but it looks like a UUID. Here the session is: **875030865e8041ac9876846271b206ac**

**Client -> Server**
RECORD rtsp://127.0.0.1:9100/mypath RTSP/1.0
Range: npt=0.000-
CSeq: 5
User-Agent: Lavf60.16.100
Session: 875030865e8041ac9876846271b206ac

**Server -> Client**
RTSP/1.0 200 OK
CSeq: 5
Server: gortsplib
Session: 875030865e8041ac9876846271b206ac

The RTSP session ID negotiated at the beginning of the connection (captured by the proxy) differs from the session ID managed by the MediaMTX server. Both the API and the logs show the same identifier, one of them truncated but they are the same id. Here the sessions are: **f254d970-cecb-4d5e-bdfd-3b28046b3c5d** and **f254d970**.

**API response for: http://localhost:9997/v3/rtspsessions/list**
```
{
    "itemCount": 1,
    "pageCount": 1,
    "items": [
        {
            "id": "f254d970-cecb-4d5e-bdfd-3b28046b3c5d",
            "created": "2024-05-15T06:34:57.661889163Z",
            "remoteAddr": "[::1]:58216",
            "state": "publish",
            "path": "mypath",
            "transport": "TCP",
            "bytesReceived": 31535526,
            "bytesSent": 0
        }
    ]
}
```

**MediaMTX logs:**
```
2024/05/15 06:33:52 INF MediaMTX v1.3.1
2024/05/15 06:33:52 INF configuration loaded from /mediamtx.yml
2024/05/15 06:33:52 INF [metrics] listener opened on 127.0.0.1:9998
2024/05/15 06:33:52 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2024/05/15 06:33:52 INF [RTMP] listener opened on :1935
2024/05/15 06:33:52 INF [HLS] listener opened on :8888
2024/05/15 06:33:52 INF [WebRTC] listener opened on :8889 (HTTP), :8189 (ICE/UDP)
2024/05/15 06:33:52 INF [SRT] listener opened on :8890 (UDP)
2024/05/15 06:33:52 INF [API] listener opened on 127.0.0.1:9997
2024/05/15 06:34:48 INF [RTSP] [conn [::1]:58216] opened
2024/05/15 06:34:57 INF [RTSP] [session f254d970] created by [::1]:58216
2024/05/15 06:35:14 INF [RTSP] [session f254d970] is publishing to path 'mypath', 2 tracks (H264, MPEG-4 Audio)
2024/05/15 06:34:48 INF [RTSP] [conn [::1]:58216] opened
...
```

This discrepancy causes issues in session tracking and logging.

**Questions:**
1. Is this a bug or are there two different session concepts that I am mixing up?
2. Is there a specific configuration or code adjustment needed to ensure the consistency of session IDs between the session used in the protocol and the one used in MediaMTX?

**Comment by aler9** [comment, marked as answer]
Hello,

> are there two different session concepts that I am mixing up?

> Is there a specific configuration or code adjustment needed to ensure the consistency of session IDs between the session used in the protocol and the one used in MediaMTX?

Exactly, there are two different session concepts. The value you are seeing in the RTSP Session header is not the session ID, but the session secret - that's a private value that is never publicized because it might allow someone to steal the session from another user. We won't make it public.

An alternative solution which might fix your problem consist into adding an additional field called "ID" to RTSP responses, in order to allow matching public session IDs with RTSP sessions. If you're interested into this, create a Feature Request.
